src/odb/include/odb/db.h

```diff
@@ -7579,12 +7575,30 @@ class dbPowerSwitch : public dbObject
   // User Code Begin dbPowerSwitch
   static dbPowerSwitch* create(dbBlock* block, const char* name);
   static void destroy(dbPowerSwitch* ps);
-  void setInSupplyPort(const std::string& in_port);
-  void setOutSupplyPort(const std::string& out_port);
+  void addInSupplyPort(const std::string& in_port);
+  void addOutSupplyPort(const std::string& out_port);
   void addControlPort(const std::string& control_port);
   void addOnState(const std::string& on_state);
+  void setLibCell(dbMaster* master);
+  void addPortMap(const std::string& model_port,
+                  const std::string& switch_port);
+
+  void addPortMap(const std::string& model_port, dbMTerm* mterm);
   std::vector<std::string> getControlPorts();
+  std::vector<std::string> getInputSupplyPorts();
+  std::vector<std::string> getOutputSupplyPorts();
   std::vector<std::string> getOnStates();
+
+  // Returns all lib cells that were defined in the upf for this power switch
+  // In case of multiple cells, the first one is the cell associated with the
+  // dbMTerms defined
+  //  in the port map
+  dbMaster* getLibCell();
```

Comment:
The comment talks about returning multiple cells but the api just a single cell.  What is intended here?

Reply:
I apologize, the comment was outdated, and I've fixed it now